gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -210,7 +211,12 @@ public static ClientContext create(StubSettings settings) throws IOException {
     if (transportChannelProvider.needsExecutor() && settings.getExecutorProvider() != null) {
       transportChannelProvider = transportChannelProvider.withExecutor(backgroundExecutor);
     }
-    Map<String, String> headers = getHeadersFromSettings(settings);
+    CredentialTypeForMetrics credentialTypeForMetrics =
+        credentials == null
+            ? CredentialTypeForMetrics.DO_NOT_SEND
+            : credentials.getMetricsCredentialType();
+    Map<String, String> headers =
+        getHeadersFromSettingsAndAppendCredentialType(settings, credentialTypeForMetrics);
```

Comment:
This method can just be `getHeaders`, where we can pass `credentials` into the method, and encapsulate all credentials metric related logic into one method.